### PR TITLE
Fix comparison tables bug

### PR DESCRIPTION
**Pull request by y3l2n**
Should fix the [bug] not returning comparison tables for objectFormat.

- fixed objectFormat types
- added COMPARISON tables to tests
- fixed bug in objectFormat parsing

[bug]: https://github.com/googledatastudio/tooling/issues/140

### CODE REVIEW

File: packages/ds-component/src/types.ts [hunk 1]
```diff
@@ -679,8 +679,8 @@ export interface ObjectRow {
 
 export interface ObjectTables {
   [TableType.DEFAULT]: ObjectRow[];
-  [TableType.COMPARISON]?: ObjectRow[][];
-  [TableType.SUMMARY]?: ObjectRow[][];
+  [TableType.COMPARISON]?: ObjectRow[];
```
Comment (diminishedprime) [inline packages/ds-component/src/types.ts]: These types _might_ be right. I'd double check after updating based on my other comment.

File: packages/ds-component/src/index.ts [hunk 2]
```diff
@@ -211,6 +211,7 @@ const objectFormatTable = (message: Message): ObjectTables => {
       const current = objectTables[table.id];
       if (current === undefined) {
         objectTables[table.id] = [];
+        objectTables[table.id] = objectRows;
```
[the remark this answers is not in the document]
Comment (y3l2n) [reply packages/ds-component/src/index.ts]: fixed

File: packages/ds-component/src/types.ts (hunk 1 above)
Comment (y3l2n) [reply packages/ds-component/src/types.ts]: Pretty sure this is right. I don't remember why those would be different.

**Comment by y3l2n** [comment]
I upgraded typescript and everything passes, and the tests use a message that aligns with what we've promised in the documentation, so I'm going to stick with these types.

**Pull request merged**